simple_salesforce/tests/test_bulk.py

```diff
@@ -0,0 +1,565 @@
+        )
+        data = [{
+            'id': 'ID-1',
+        }, {
+            'id': 'ID-2',
+        }]
+        session = requests.Session()
+        client = Salesforce(session_id=tests.SESSION_ID,
+                            instance_url=tests.SERVER_URL,
+                            session=session)
+        contact = client.bulk.Contact.delete(data)
+        self.assertEqual([self.expected], contact)
+
+    @responses.activate
+    def test_insert(self):
+        """Test bulk insert records"""
+        operation = 'insert'
+        responses.add(
+            responses.POST,
+            re.compile(r'^https://[^/job].*/job$'),
+            body='{"apiVersion": 42.0, "concurrencyMode": "Parallel",'
+            '"contentType": "JSON","id": "Job-1","object": "Contact",'
+            '"operation": "%s","state": "Open"}' % operation,
+            status=http.OK)
+        responses.add(
+            responses.POST,
+            re.compile(r'^https://[^/job].*/job/Job-1/batch$'),
+            body='{"id": "Batch-1","jobId": "Job-1","state": "Queued"}',
+            status=http.OK
+        )
+        responses.add(
+            responses.POST,
+            re.compile(r'^https://[^/job].*/job/Job-1$'),
+            body='{"apiVersion" : 42.0, "concurrencyMode" : "Parallel",'
+            '"contentType" : "JSON","id" : "Job-1","object" : "Contact",'
+            '"operation" : "%s","state" : "Closed"}' % operation,
+            status=http.OK
+        )
+        responses.add(
+            responses.GET,
+            re.compile(r'^https://[^/job].*/job/Job-1/batch/Batch-1$'),
+            body='{"id": "Batch-1","jobId": "Job-1","state": "InProgress"}',
+            status=http.OK
+        )
+        responses.add(
+            responses.GET,
+            re.compile(r'^https://[^/job].*/job/Job-1/batch/Batch-1$'),
+            body='{"id": "Batch-1","jobId": "Job-1","state": "Completed"}',
+            status=http.OK
+        )
+        responses.add(
+            responses.GET,
+            re.compile(
+                r'^https://[^/job].*/job/Job-1/batch/Batch-1/result$'),
+            body='[{"success": true,"created": true,"id": "001xx000003DHP0AAO",'
+            '"errors": []},{"success": true,"created": true,'
+            '"id": "001xx000003DHP1AAO","errors": []}]',
+            status=http.OK
+        )
+
+        data = [{
+            'AccountId': 'ID-1',
+            'Email': 'contact1@example.com',
+            'FirstName': 'Bob',
+            'LastName': 'x'
+        }, {
+            'AccountId': 'ID-2',
+            'Email': 'contact2@example.com',
+            'FirstName': 'Alice',
+            'LastName': 'y'
+        }]
+        session = requests.Session()
+        client = Salesforce(session_id=tests.SESSION_ID,
+                            instance_url=tests.SERVER_URL,
+                            session=session)
+        contact = client.bulk.Contact.insert(data)
+        self.assertEqual([self.expected], contact)
```

Comment:
I added `[self.expected]` to insert,update, upsert and so on because of https://github.com/simple-salesforce/simple-salesforce/pull/259/files#diff-0f8e08704aa000282b5b1703b2c78af3L226. I wonder if it would be better to use `extend`.